daemons/fenced/fenced_remote.c

```diff
@@ -1484,33 +1484,59 @@ stonith_choose_peer(remote_fencing_op_t * op)
     return NULL;
 }
 
+static int
+valid_fencing_timeout(int specified_timeout, const char *scope,
+                      const remote_fencing_op_t *op, const char *device)
+{
+    int timeout = specified_timeout;
+
+    if (!is_watchdog_fencing(op, device)) {
+        return timeout;
+    }
+
+    timeout = (int) QB_MIN(QB_MAX(specified_timeout,
+                                  stonith_watchdog_timeout_ms / 1000), INT_MAX);
+
+    if (timeout > specified_timeout) {
+        crm_warn("Specified '%s' fencing timeout %ds is too short (must be >= "
+                  PCMK_OPT_STONITH_WATCHDOG_TIMEOUT " %ds), using %ds instead",
+                  scope, specified_timeout, timeout, timeout);
```

Comment:
To help the user, let's say `pcmk_%s_timeout for %s` (device name). For the non-action timeouts, we can pass NULL as the scope and use a more generic warning.

Reply:
Makes sense. Let me know if it could be anyhow even better .

Testing is looking good AFAICS.